### PR TITLE
Adds MTLS support to Traefik chart.

### LINKED ISSUE #9828: [stable/traefik] Support mutual TLS authentication
**Is this a request for help?**:

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): FEATURE REQUEST

**Version of Helm and Kubernetes**:

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.0", GitCommit:"ddf47ac13c1a9483ea035a79cd7c10005ff21a6d", GitTreeState:"clean", BuildDate:"2018-12-04T07:51:55Z", GoVersion:"go1.11.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.11", GitCommit:"637c7e288581ee40ab4ca210618a89a555b6e7e9", GitTreeState:"clean", BuildDate:"2018-11-26T14:25:46Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}

$ helm version
Client: &version.Version{SemVer:"v2.11.0", GitCommit:"2e55dbe1fdb5fdb96b75ff144a339489417b146b", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.11.0", GitCommit:"2e55dbe1fdb5fdb96b75ff144a339489417b146b", GitTreeState:"clean"}
```

**Which chart**: stable/traefik

**What happened**: Unable to configure traefik to enable [mutual TLS authentication](https://docs.traefik.io/configuration/entrypoints/#tls-mutual-authentication)

**What you expected to happen**: The chart should provide the ability to enable and configure mTLS

**How to reproduce it** (as minimally and precisely as possible):

1. Deploy traefik
2. Deploy a HTTPS web server that echoes received headers back to the client, such as [echoheaders](https://github.com/kubernetes/contrib/tree/master/ingress/echoheaders)
3. Access the web server, providing a client certificate: `curl --cert tls.crt --key tls.key https://myserver.example.com`
4. Notice there is no "x-forwarded-tls-client-cert" header in the returned response body
5. Notice that it is possible to access the web server without using mTLS: `curl https://myserver.example.com`

**Anything else we need to know**:
No

### PR DESCRIPTION
**Pull request by tlm**
#### What this PR does / why we need it:
Adds support to the Traefik helm chart to support its already existing MTLS settings through helm values. This support is needed so Traefik can be deployed into Kubernetes using helm and support mutual TLS auth.

#### Which issue this PR fixes
#9828 

#### Special notes for your reviewer:
The original work was done by @somcsel

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by tpdownes** [comment]
This commit is the Lord's work. I literally need it today! Thank you. 2 comments:

1. When I install, delete + purge, and install again it tells me that `Error: jobs.batch "storeconfig-job-1" already exists`. I'm not sure if I'm doing something wrong but maybe there's a minor tweak to recipe that would solve this?

2. Confusing docs regarding how to specify the CA certificates.

```
ssl.mtls.enabled is set, an array of client CA certificates, each in Base64 encoded PEM format
```

PEM files are already a particular form of base64 encoding! What the code expects me to do something like `cat cacert.pem | base64` and put that into the file. So, what you want me to do is base64 encode a base64 encoded file! This might be semi-normal to people used to creating k8s secrets, but it's funny wording. You should either clarify that or consider alternative like:

```
ssl:
  enabled: true
  enforced: true
  permanentRedirect: false
  mtls:
    enabled: true
    optional: true
    clientCaCerts: |
      -----BEGIN CERTIFICATE-----
      MIID1TCCAr2gAwIBEBBQUAMGoxEzARBgoJ
      ...
      Z6Vku+8ShpB71NWapBjsS9vUQ==
      -----END CERTIFICATE-----
```
(where I've truncated and garbled a real cert)

**Comment by tlmiller** [comment]
@tpdownes cheers for the comments.

1. Isn't related to this PR it's a helm specific things where object installed into a cluster with a pre hook are not automatically cleaned up with a helm delete. If you look at https://github.com/helm/charts/blob/master/stable/traefik/templates/storeconfig-job.yaml you can see that it has an annotation to tell helm about the hook.

2. Thanks for the feedback. You are correct that is confusing, I am going to get rid of the base64 requirement so the values read better and it's inline with the rest of the charts secrets.

**Comment by tlmiller** [comment]
@ldez @dtomcej is it possible to get another reviewer to take a look at this? Cheers

File: stable/traefik/values.yaml [hunk 1]
```diff
@@ -88,6 +88,13 @@ ssl:
   auth: {}
     # basic:
     #   testuser: $apr1$JXRA7j2s$LpVns9vsme8FHN0r.aSt11
+  mtls:
```
Comment (dtomcej) [inline stable/traefik/values.yaml:91]: Can you make this optional (commented out) or with an empty object (as with `auth: {}` above?)

**Review by dtomcej — COMMENTED** [review]
Hello!

One small change, and a version bump if needed!

File: stable/traefik/values.yaml (hunk 1 above)
Comment (tlm) [reply stable/traefik/values.yaml:91]: I am happy to do this. However does not make sense to keep it the way it is with an enabled flag? This seems to be inline with the rest of the chart.

**Pull request closed without merging**